design/animation.md

```diff
@@ -0,0 +1,317 @@
+can be created by explicitly binding an animation (`Animation.bind(...objects)`), or may be returned during actor
+creation from `Actor.CreateFromPrefab`/`Gltf`/`Library`. Unlike the asset, instances can be played, paused, set speed,
+etc. Animation instances can either be stored directly, or obtained from any of the actors/materials that participate
+in the animation. Instances participate in the patching system, so changing properties on the animation instance
+are pushed down to clients automatically.
+
+An animation instance can be set to start another animation on the instant that it finishes, by setting an instance's
+`continueWith` property. Because it's set in advance, there is zero downtime between one ending and another starting,
+and does not require a round trip.
+
+```ts
+class Animation {
+	bind(...args: Animatible[]): AnimationInstance;
+}
+
+class AnimationInstance {
+	/** Generated unique ID */
+	id: Readonly<string>;
+
+	/** Playback speed multiplier */
+	speed: number;
+	/** What happens when the animation hits the last frame */
+	wrapMode: WrapMode;
+	/** When multiple animations play together, this is the relative strength of this instance */
+	weight: number;
+	/** The current playback time, based on start time and speed */
+	time: number;
+	/** The list of actors/materials that participate in this instance */
+	targets: Readonly<Animatible[]>;
+	/** The playing animation */
+	animation: Readonly<Animation>;
+	/** Start these animation instances after this one completes */
+	continueWith: AnimationInstance[];
+
+	/** Current play state */
+	isPlaying: Readonly<boolean>;
+
+	play(): void;
+	pause(): void;
+
+	/** Only fired if wrapMode is Once */
+	finished(): Promise<void>;
+	/** Sets the `continueWith` property, and returns the next anim instance */
+	continueWith(otherAnim: AnimationInstace): AnimationInstance;
+}
+
+enum WrapMode {
+	Once = 'once',
+	Loop = 'loop',
+	Yoyo = 'yoyo'
+}
+
+const ticktock: MRE.Animation = assets.createAnimation(...);
+const animInstance: MRE.AnimationInstance = ticktock.bind(firstBall, lastBall);
+animInstance.speed = 1.5;
+animInstance.play();
+
+// this prefab has an animation instance pre-allocated
+const cradle = MRE.Actor.CreateFromPrefab(context, { prefab: assets.prefabs[0] });
+await cradle.created();
+cradle.animations('rock').play();
+```
+
+### Convenience function
+
+For many purposes, creating and managing an animation is overkill. This is why objects that can be animated have a
+convenient fire-and-forget function: `animateTo`. This function takes a "destination" ActorLike or MaterialLike and
+a duration as input, generates a simple one-frame animation from all the fields given in the -Like, binds it to the
+object, and plays it.
+
+```ts
+class Animation {
+	public static AnimateTo(target: Animatible, options: {
+		duration: number;
+		destination: ActorLike | MaterialLike;
```

Comment:
I think this needs to be `Partial<ActorLike> | Partial<MaterialLike>`, so that users don't have to pass a fully filled out value. It will make statically validating them impossible, but would alleviate the need for users to cast them, e.g. `as MaterialLike`.